<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigWidget</class>
 <widget class="QWidget" name="ConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>685</width>
    <height>627</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Build Directory:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="KUrlRequester" name="buildDir">
       <property name="toolTip">
        <string>Select the directory where the tools are being executed in.
If this is the same as the source directory you can leave it empty.</string>
       </property>
       <property name="mode">
        <set>KFile::Directory|KFile::ExistingOnly|KFile::LocalOnly</set>
       </property>
       <property name="clickMessage">
        <string>The directory where the tools are executed</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Build Tools</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Action:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KComboBox" name="buildAction"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Enable:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="enableAction">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="execLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Executable:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="KUrlRequester" name="actionExecutable">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>This is the executable that should be executed when selecting the action from the Project menu</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="argLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Arguments:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="actionArguments">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>The arguments that the tool needs to execute properly.</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="envLabel">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Environment:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="KDevelop::EnvironmentSelectionWidget" name="actionEnvironment">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Choose a pre-configured environment for your tool</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Includes/Imports and Defines for Project Directories/Files</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Directories/Files:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QListView" name="projectPaths">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>3</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>A list of paths specifying project directories or files, for which includes and
defines should be set up. &quot;/&quot; denotes the top level project directory. If the
path identifies a directory the includes/defines are active for that directory
and all sub (and sub-sub)directories.</string>
        </property>
        <property name="whatsThis">
         <string/>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="KComboBox" name="switchIncludeDefines">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Switch between configuring Includes/Imports and Defines</string>
        </property>
        <item>
         <property name="text">
          <string>Includes/Imports</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Defines</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QStackedWidget" name="stackedWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>8</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>List of include paths for the selected project item</string>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="page">
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="margin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QListView" name="includePaths">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>8</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>A list of absolute include paths that the KDevelop parser should be using
when parsing the project files in the given project path.</string>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="margin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTableView" name="defines">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>8</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Specify a list of pre-processor defines and their values that the KDevelop parser should use when parsing the files in the given project path.</string>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KDevelop::EnvironmentSelectionWidget</class>
   <extends>QComboBox</extends>
   <header>util/environmentselectionwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionExecutable</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionArguments</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>142</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionEnvironment</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>103</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>execLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>51</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>argLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>35</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>envLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>34</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>switchIncludeDefines</sender>
   <signal>activated(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>459</y>
    </hint>
    <hint type="destinationlabel">
     <x>685</x>
     <y>479</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
